third-party/jemalloc/Makefile

```diff
@@ -15,7 +15,8 @@ CHPL_JEMALLOC_CFG_OPTIONS += --host=$(CHPL_MAKE_TARGET_PLATFORM)-unknown-linux-g
 endif
 
 CHPL_JEMALLOC_CFG_OPTIONS += --prefix=$(JEMALLOC_INSTALL_DIR) \
-			     --with-jemalloc-prefix=je_
+                             --with-jemalloc-prefix=je_ \
+                             --with-private-namespace=jeint_
```

Comment:
Understood that the need for this comes from the current PR, but still it seems it might be better as a separate one done before the current one.  Perhaps with a mention that the current one is what creates the need.

Also, "jeint" seems vague and/or subject to collision with someone else doing the same thing.  Maybe "jechpl"?  (I could actually see using "jechpl" for both the public and private symbols, but realize "je" for the public ones predates this change.  @ronawho may have some input here also.)